storage/src/mmr/journaled.rs

```diff
@@ -522,6 +524,7 @@ impl<E: RStorage + Clock + Metrics, H: CHasher> Mmr<E, H> {
         }
 
         self.journal.rewind(new_size).await?;
+        self.journal.sync().await?;
```

Comment:
I was going to comment that we may not want to call `sync` "per pop" but it looks like it already takes a number to walk back, so looks good 👍